### PR TITLE
To Do list

### PR DESCRIPTION
**Pull request by ridaarif98**
In this todo list,
- Set up a  project with webpack 
- Create an index.html file and add HTML markup 
- Create an empty To-Do List placeholder
- Create an index.js file and set an array of some simple to do tasks 
- Each task object should contain three keys:
  i- description [string].
  ii- completed [bool].
  iii- index: [number]
- Integrate linters
- Fix linter  errors
- Write a function to iterate over the tasks array and populate an HTML list item element for each task.
- On page load render the dynamically created list of tasks in the dedicated placeholder
- Create a style.css and set rules for the To Do List

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,47 @@
...
+    complete: false,
+  };
+  collection.push(todo);
+  localStorage.setItem('todoList', JSON.stringify(collection));
+  if (collection.length > 0) {
+    ShowList(collection);
+  }
+  form.reset();
+}
+
```
Comment (arslanbisharat) [inline src/index.js]: - [ ] Please make sure it has an array of some simple to-do tasks (array of objects) with each object containing three keys - description [string], completed [bool], index: [number]. You can choose to have more properties or name these properties differently.
```javascript
const tasks = [
  {
    description: 'An example description in string',
    completed: false,
    index: 10,
  },
];
```

File: src/index.js [hunk 2]
```diff
@@ -1 +1,47 @@
...
+  const data = JSON.parse(dataGet);
+  if (data) {
+    collection = data;
+  }
+  if (collection.length > 0) {
+    ShowList(collection);
+  }
+});
+
+form.addEventListener('submit', (ev) => {
```
Comment (arslanbisharat) [inline src/index.js]: - [ ] Please make sure it has a function that iterates over the tasks array and populates the HTML list item element for each task.

```javascript
const populateTasks = (task, parentElement) => {
  const { description, completed, index: id } = task;
  const innerHTML = `
         <p>${description}</p>
         <p>${completed ? 'completed' : 'not completed'}</p>`;
  const newTodoListElement = document.createElement('li');
  Object.assign(newTodoListElement, { innerHTML, id });
  parentElement.appendChild(newTodoListElement);
};
```

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js (hunk 2 above)
Comment (ridaarif98) [reply src/index.js]: thanks for your review but I used a function to iterate over the array that is show list and I used a map inside the function. And it is working

Comment (arslanbisharat) [reply src/index.js]: If you feel it's working please accept my apologies. You can request another review was you feel you are done with all the changes that can implement. Thank you for understanding

Comment (ridaarif98) [reply src/index.js]: Thank you for your reply. I will update my project with required changes and submit again

**Review by mikenath223 — APPROVED** [review]
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 3]
```diff
@@ -1 +1,49 @@
...
+
+function ShowList(arr) {
+  const listToDo = arr.map((b) => `
+    <ul class="testList1" draggable="true">
+          <li><input type="checkbox" id='${b.index}' value='${b.completed}' class="checkboX">
+          <p>${b.description}</p>
+          </li>
+          <i class=" fa fa-ellipsis-v" id='${b.index}'></i>
+          <i class="fa fa-trash-o" id='${b.index}'></i>          
+      </ul>
```
Comment (mikenath223) [inline src/index.js:31]: - [ ] OPTIONAL - Consider changing this regular function to es6 arrow function. As stated in the [Javascript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md#:~:text=If%20you%20use%20ES6%2C%20use%20object%20destructuring%20to%20get%20the%20values%20from%20an%20object.%20This%20way%20you%20can%20avoid%20repeating%20a%20lot%20of%20code.).

**Pull request merged**